server/src/test/java/io/spine/examples/chatspn/server/message/MessageSendingTest.java

```diff
@@ -0,0 +1,121 @@
+import io.spine.examples.chatspn.message.sendingrejection.SendingRejections.MessageCannotBeSent;
+import io.spine.examples.chatspn.server.ChatsContext;
+import io.spine.server.BoundedContextBuilder;
+import io.spine.testing.core.given.GivenUserId;
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.examples.chatspn.server.given.MessageSendingTestEnv.createRandomChat;
+import static io.spine.examples.chatspn.server.given.MessageSendingTestEnv.sendMessage;
+
+@DisplayName("`MessageSending` should")
+public final class MessageSendingTest extends ContextAwareTest {
+
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return ChatsContext.newBuilder();
+    }
+
+    @Test
+    @DisplayName("emit `MessageSent` event")
+    void event() {
+        Chat chat = createRandomChat(context());
+        Message message = sendMessage(chat.getId(),
+                                      chat.getMember(0),
+                                      context());
+
+        MessageSent expectedEvent = MessageSent
+                .newBuilder()
+                .setId(message.getId())
+                .setChat(message.getChat())
+                .setUser(message.getUser())
+                .setContent(message.getContent())
+                .vBuild();
+
+        context().assertEvents()
+                 .withType(MessageSent.class)
+                 .hasSize(1);
+        context().assertEvents()
+                 .withType(MessageSent.class)
+                 .message(0)
+                 .isEqualTo(expectedEvent);
+    }
+
+    @Test
+    @DisplayName("post message")
+    void state() {
+        Chat chat = createRandomChat(context());
+        Message message = sendMessage(chat.getId(),
+                                      chat.getMember(0),
+                                      context());
+        context().assertState(message.getId(), Message.class)
+                 .comparingExpectedFieldsOnly()
+                 .isEqualTo(message);
+    }
+
+    @Test
+    @DisplayName("reject when the message sender is not the chat member")
+    void rejection() {
+        Chat chat = createRandomChat(context());
+        Message message = sendMessage(chat.getId(),
+                                      GivenUserId.generated(),
+                                      context());
+
+        MessageCannotBeSent expectedRejection = MessageCannotBeSent
+                .newBuilder()
+                .setId(message.getId())
+                .setChat(message.getChat())
+                .setUser(message.getUser())
+                .setContent(message.getContent())
+                .vBuild();
+
+        context().assertEvents()
+                 .withType(MessageCannotBeSent.class)
+                 .message(0)
+                 .isEqualTo(expectedRejection);
+    }
+
+    @Test
+    @DisplayName("archive itself after work")
```

Comment:
After any "work"? Or after, say, successful completion (once the message is sent)?